index.ts

```diff
@@ -248,6 +278,28 @@ export const tokenize = (
             done
         }
 
+        const unicodeState = (): State => {
```

Comment:
add a comment to this function to explain what it tries to do, otherwise looks good